opm/autodiff/SimulatorFullyImplicitBlackoilOutput.cpp

```diff
@@ -252,6 +254,45 @@ namespace Opm
         }
     }
 
+    struct WriterCall
+    {
+        BlackoilOutputWriter& writer_;
+        std::unique_ptr< SimulatorTimerInterface > timer_;
+        const SimulatorState state_;
+        const WellState wellState_;
+        const bool substep_;
+
+        std::future< bool > asyncWait_;
```

Comment:
I admit this is too complicated for me to comment on quickly, my idea was adding the future to the BlackoilOutputWriter class, not the WriterCall (I guess I was unclear). Did you try that? I imagine it would be simpler.